fairlearn/metrics/_group_metric_result.py

```diff
@@ -1,8 +1,16 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
```

Comment:
This class right now feels like an overkill. I think I'm in favor of using the lighterweight approach that we played with a while back. Also, my proposal below should handle vector-valued metrics (but we can drop those and simplify things even further):

```python
from sklearn.utils import Bunch
import numpy as np

class GroupMetricResult(Bunch):

    def __init__(self, overall, by_group):
        self.overall = overall
        self.by_group = by_group

        self.min = np.min(list(by_group.values()), axis=0)   # or: min(by_group.values())
        self.max = np.max(list(by_group.values()), axis=0)   # or: max(by_group.values())
        self.disparity_diff = self.max - self.min
        self.disparity_ratio = _vec_positive_divide(self.min, self.max) # or: _positive_divide(self.min, self.max)

def _positive_divide(small, large):
    if (small<0):
        return np.nan
    elif (large==0):
        return 1.0
    else:
        return small/large

_vec_positive_divide = np.frompyfunc(_positive_divide, 2, 1)
```

Reply:
Wouldn't deriving from `Bunch` make this a _heavier_ class?

Also in the case of vector-valued metrics, isn't `numpy.minimum` going to mix together elements from different subgroups? If so, does that really leave the `minimum` meaning anything?

Reply:
why was this merged? let's make sure to resolve the comments before merging in future.